src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/MessageboxInstancesController.cs

```diff
@@ -179,6 +180,107 @@ public class MessageBoxInstancesController : ControllerBase
             return Ok(authorizedInstances);
         }
 
+        /// <summary>
+        /// Search through instances to find match based on query params.
+        /// </summary>
+        /// <param name="queryModel">Object with query-params</param>
+        /// <returns>list of instances</returns>
+        [Authorize]
+        [HttpPost("search")]
+        public async Task<ActionResult> SearchMessageBoxInstances([FromBody] MessageBoxQueryModel queryModel)
+        {
+            string[] acceptedLanguages = { "en", "nb", "nn" };
+
+            string languageId = "nb";
+
+            if (queryModel.Language != null && acceptedLanguages.Contains(queryModel.Language.ToLower()))
+            {
+                languageId = queryModel.Language.ToLower();
+            }
+
+            Dictionary<string, StringValues> queryParams = GetQueryParams(queryModel);
```

Comment:
query params are not used before line 221. Could we do this mapping at that point instead? We might not need if it we return an empty list in line 213.

Reply:
Good point.